### PR TITLE
Package tablecloth-native.0.0.2

### PR DESCRIPTION
**Pull request by pbiggar**
### `tablecloth-native.0.0.2`
Native OCaml library implementing Tablecloth, a cross-platform standard library for OCaml, Bucklescript and ReasonML
Longer description

---
* Homepage: https://github.com/darklang/tablecloth
* Source repo: git://github.com/darklang/tablecloth
* Bug tracker: https://github.com/darklang/tablecloth/issues

---
:camel: Pull-request generated by opam-publish v2.0.0

### CODE REVIEW

**Comment by pbiggar** [comment]
Are there logs for camelus? How do I debug this?

**Comment by kit-ty-kate** [comment]
Unfortunately there is no detailed log for Camelus but the problems are very obvious here: the package can't be installed because `base >= v10.0` doesn't exist. You probably mean `base >= v0.10.0` instead.

File: packages/tablecloth-native/tablecloth-native.0.0.2/opam [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  "Native OCaml library implementing Tablecloth, a cross-platform standard library for OCaml, Bucklescript and ReasonML"
+description: "Longer description"
+maintainer: "Paul Biggar <paul@darklang.com>"
+authors: "Paul Biggar <paul@darklang.com>"
+license: "MIT with some exceptions"
+homepage: "https://github.com/darklang/tablecloth"
+bug-reports: "https://github.com/darklang/tablecloth/issues"
+depends: [
+  "ocaml"
+  "ocamlfind"
```
Comment (kit-ty-kate) [inline packages/tablecloth-native/tablecloth-native.0.0.2/opam]: This dependency doesn't seem to be used anywhere in your source code. Could you remove it?

File: packages/tablecloth-native/tablecloth-native.0.0.2/opam [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  "dune"
```
Comment (kit-ty-kate) [inline packages/tablecloth-native/tablecloth-native.0.0.2/opam]:
```suggestion
  "dune" {build}
```
This dependency should be tagged with the `{build}` attribute

File: packages/tablecloth-native/tablecloth-native.0.0.2/opam [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  "base" {>= "v10.0"}
+  "str"
+]
+build: ["dune" "build" "-p" name]
```
Comment (kit-ty-kate) [inline packages/tablecloth-native/tablecloth-native.0.0.2/opam]:
```suggestion
build: ["dune" "build" "-p" name "-j" jobs]
```
The idiomatic call to dune usually have also `"-j" jobs` so than people can parallelize (or not) as they wish

File: packages/tablecloth-native/tablecloth-native.0.0.2/opam [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+maintainer: "Paul Biggar <paul@darklang.com>"
+authors: "Paul Biggar <paul@darklang.com>"
+license: "MIT with some exceptions"
+homepage: "https://github.com/darklang/tablecloth"
+bug-reports: "https://github.com/darklang/tablecloth/issues"
+depends: [
+  "ocaml"
+  "ocamlfind"
+  "dune"
+  "base" {>= "v10.0"}
```
Comment (kit-ty-kate) [inline packages/tablecloth-native/tablecloth-native.0.0.2/opam]:
```suggestion
  "base" {>= "v0.10.0"}
```

File: packages/tablecloth-native/tablecloth-native.0.0.2/opam (hunk 1 above)
Comment (pbiggar) [reply packages/tablecloth-native/tablecloth-native.0.0.2/opam]: 👍

File: packages/tablecloth-native/tablecloth-native.0.0.2/opam (hunk 2 above)
Comment (pbiggar) [reply packages/tablecloth-native/tablecloth-native.0.0.2/opam]: 👍

File: packages/tablecloth-native/tablecloth-native.0.0.2/opam (hunk 3 above)
Comment (pbiggar) [reply packages/tablecloth-native/tablecloth-native.0.0.2/opam]: 👍

File: packages/tablecloth-native/tablecloth-native.0.0.2/opam (hunk 4 above)
Comment (pbiggar) [reply packages/tablecloth-native/tablecloth-native.0.0.2/opam]: 👍

**Comment by pbiggar** [comment]
Thanks for the review and for catching that base version problem! Fixed them all and resubmitted.

File: packages/tablecloth-native/tablecloth-native.0.0.2/opam [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+maintainer: "Paul Biggar <paul@darklang.com>"
+authors: "Paul Biggar <paul@darklang.com>"
+license: "MIT with some exceptions"
+homepage: "https://github.com/darklang/tablecloth"
+bug-reports: "https://github.com/darklang/tablecloth/issues"
+depends: [
+  "ocaml"
+  "dune" {build}
+  "base" {>= "v0.10.0"}
+  "str"
```
Comment (kit-ty-kate) [inline packages/tablecloth-native/tablecloth-native.0.0.2/opam]:
```suggestion
```
There is still a problem. This package doesn't exists. My guess is that you wanted to tell that you are using the `Str` module. However this is a builtin module installed by the compiler.

Comment (pbiggar) [reply packages/tablecloth-native/tablecloth-native.0.0.2/opam]: I added this in response to a build failure on this CI. I can't find the log but the error was "No implementations provided for the following modules: Str", and stackoverflow suggested this: https://stackoverflow.com/questions/38547815/no-implementations-provided-for-the-following-modules-str.

I'll remove it and see what happens and we can debug from there.

Thanks again :)

Comment (kit-ty-kate) [reply packages/tablecloth-native/tablecloth-native.0.0.2/opam]: str is vendored as a separate library even if it's installed by the compiler. You need to add it explicitly in your dune file for it to be linked

Comment (pbiggar) [reply packages/tablecloth-native/tablecloth-native.0.0.2/opam]: Gotcha, thanks! That seems to have fixed it :)

**Pull request merged**